components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ChoreoApiMgtDAO.java

```diff
@@ -266,6 +268,39 @@ public Map<String, String> getEnvironmentToGatewayAccessibilityTypeMapping(Set<S
         return envToGatewayAccessibilityTypeMap;
     }
 
+    public Map<String, String> getEnvDetailsForDeployEvent(String environment, String organization)
+        throws APIManagementException {
+        Map<String, String> envDetailsMap = new HashMap<>();
+        if (environment == null || environment.equals("")) {
+            return envDetailsMap;
+        }
+        Map<String, Environment> readOnlyEnvironments = APIUtil.getReadOnlyEnvironments();
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            try (PreparedStatement prepStmt =
+                         connection.prepareStatement(SQLConstants.GET_ENVIRONMENT_DETAILS_MAPPING_SQL)) {
+                prepStmt.setString(1, environment);
+                prepStmt.setString(2, organization);
+                try (ResultSet rs = prepStmt.executeQuery()) {
+                    if (rs.next()) {
+                        envDetailsMap.put("dataPlaneID", rs.getString("DATA_PLANE_ID"));
+                        envDetailsMap.put("gatewayAccessibilityType", rs.getString("ACCESSIBILITY_TYPE"));
+                    }
+                }
+            }
+            // check if the environment is in the read-only environments list
+            if (readOnlyEnvironments.containsKey(environment)) {
```

Comment:
shall we move this logic to line number 278, such that it would return if the condition is not met prior to doing the DB call?

Reply:
fixed via https://github.com/wso2/carbon-apimgt/commit/332523e11f9d146915b16f17ed8316c6cd82fb5e